### DISCUSSION TITLE
How do you mock Apollo GraphQL in Next.js SSR getServerSideProps?

### DISCUSSION THREAD

**Discussion by gvocale** [cypress-io/cypress#17644] [CI setup]
I have a Next.JS app that fetches via Apollo GraphQL server side via the `getServerSideProps` hook.

How can I intercept, mock, stub those requests in Cypress?

I have issue running the Cypress GitHub Action because it is failing on all these GraphQL queries happening server side. 

```tsx
import { GetServerSideProps } from "next";
import { ApolloClient, ApolloLink, InMemoryCache } from "@apollo/client";

export const getServerSideProps: GetServerSideProps = async (context) => {
  const {
    params: { slug },
  } = context;

  const client = new ApolloClient({
    ssrMode: true,
    link: ApolloLink....
    cache: new InMemoryCache(),
  });


  try {
    const { data } = await client.query<ProductBySlugQueryProps>({
      query: PRODUCT_BY_SLUG_QUERY,
      variables: { slug },
    });

    return {
      props: {
        product: data?.product || {},
      },
    };
  } catch (e) {
    captureException(e);

    return {
      notFound: true,
    };
  }
};
```

**Comment by mufasa71** [comment]
Hi, I am still new in Cypress, but have you tried [this](https://docs.cypress.io/guides/testing-strategies/working-with-graphql)?

**Reply by gvocale** [reply]
It works successfully for queries I fire via the `useQuery` Apollo GraphGL hook inside a component, but it won't work for queries happening during server side rendering inside the Next.js `getServerSideProps` hook 😐

**Reply by mufasa71** [reply]
I see. Yeah I think Cypress has nothing to do with SSR. So in this case I ended up with creating new testing backend server which runs in cloud.

**Comment by gvocale** [comment]
I managed successfully to do it via [Mock Service Worker](https://mswjs.io/)
